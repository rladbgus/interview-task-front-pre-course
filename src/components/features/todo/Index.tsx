'use client';
import React from 'react';
import styled from '@emotion/styled';
import InputForm from './InputForm';
import ListBoxForm from './ListBoxForm';

const Index = () => {
  return (
    <S.Wrapper>
      <S.Title>To Do List</S.Title>
      <InputForm />
      <ListBoxForm />
    </S.Wrapper>
  );
};

const S = {
  Wrapper: styled.div`
    margin: 100px auto;
    max-width: 700px;
    text-align: center;
  `,
  Title: styled.div`
    font-size: 56px;
    font-weight: 700;
  `
};

export default Index;
