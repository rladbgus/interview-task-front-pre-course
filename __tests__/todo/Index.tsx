import { render, screen } from '@testing-library/react';
import Index from 'components/features/todo/Index';

it('최상위 컴퍼넌트 스냅샵 테스트', () => {
  const { asFragment } = render(<Index />);
  expect(asFragment()).toMatchSnapshot();
});

it('타이틀 렌더 테스트', () => {
  render(<Index />);
  const titleElement = screen.getByText(/To Do List/i);
  expect(titleElement).toBeInTheDocument();
});
