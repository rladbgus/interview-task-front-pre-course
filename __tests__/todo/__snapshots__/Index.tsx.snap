// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`최상위 컴퍼넌트 스냅샵 테스트 1`] = `
<DocumentFragment>
  <div
    class="css-ogq495"
  >
    <div
      class="css-1byphj3"
    >
      To Do List
    </div>
    <div
      class="css-jqim4p"
    >
      <input
        class="css-1vuil63"
        placeholder="할 일을 입력해주세요"
        type="text"
        value=""
      />
    </div>
    <div
      class="css-136cwgt"
    >
      <div
        class="css-t1g06b"
      >
        <button
          class="css-1pooxy9"
        >
          All
        </button>
        <button
          class="css-1rrcqoz"
        >
          Todo
        </button>
        <button
          class="css-1rrcqoz"
        >
          Done
        </button>
      </div>
      <div
        class="css-ncewfj"
      >
         총 0개
      </div>
    </div>
  </div>
</DocumentFragment>
`;
